app/uk/gov/hmrc/partnershipidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.partnershipidentification.connectors
+
+import play.api.http.Status.OK
+import play.api.libs.json._
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.partnershipidentification.config.AppConfig
+import uk.gov.hmrc.partnershipidentification.connectors.RegisterWithMultipleIdentifiersHttpParser._
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RegisterWithMultipleIdentifiersConnector @Inject()(http: HttpClient,
+                                                         appConfig: AppConfig
+                                                        )(implicit ec: ExecutionContext) {
+
+  def register(sautr: String
+              )(implicit hc: HeaderCarrier): Future[RegisterWithMultipleIdentifiersResult] = {
+
+    val extraHeaders = Seq(
+      "Authorization" -> appConfig.desAuthorisationToken,
+      appConfig.desEnvironmentHeader,
+      "Content-Type" -> "application/json"
+    )
+
+    val jsonBody: JsObject =
+      Json.obj(
+        "partnership" ->
+          Json.obj(
+            "sautr" -> sautr
+          )
+      )
+
+    http.POST[JsObject, RegisterWithMultipleIdentifiersResult](
+      url = appConfig.getRegisterWithMultipleIdentifiersUrl,
+      headers = extraHeaders,
+      body = jsonBody
+    )(
+      implicitly[Writes[JsObject]],
+      RegisterWithMultipleIdentifiersHttpReads,
+      hc,
+      ec
+    )
+
+  }
+
+}
+
+object RegisterWithMultipleIdentifiersHttpParser {
+
+  val IdentificationKey = "identification"
+  val IdentificationTypeKey = "idType"
+  val IdentificationValueKey = "idValue"
+  val SafeIdKey = "SAFEID"
+
+  implicit object RegisterWithMultipleIdentifiersHttpReads extends HttpReads[RegisterWithMultipleIdentifiersResult] {
+
+    override def read(method: String, url: String, response: HttpResponse): RegisterWithMultipleIdentifiersResult = {
+      response.status match {
+        case OK =>
+          (for {
+            idType <- (response.json \ IdentificationKey \ 0 \ IdentificationTypeKey).validate[String]
+            if idType == SafeIdKey
+            safeId <- (response.json \ IdentificationKey \ 0 \ IdentificationValueKey).validate[String]
+          } yield safeId) match {
+            case JsSuccess(safeId, _) => RegisterWithMultipleIdentifiersSuccess(safeId)
+            case _: JsError => throw new InternalServerException(s"Invalid JSON returned on Register API: ${response.body}")
+          }
+        case _ =>
+          throw new InternalServerException(
```

Comment:
We need to return RegisterWithMultipleIdentifiersFailure instead of the Internal Server Exception, then the controller can handle it instead. Will also be good to add some tests for failures in the controller and connector